### PR TITLE
BUGFIX: SD-389 Wrong Legend in Combo chart (Area-Area)

### PR DESCRIPTION
**Pull request by nadtan**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards)
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs
- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2188
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/1883

# Related Jira tasks
[SD-389](https://jira.intgdc.com/browse/SD-389)

### CODE REVIEW

**Comment by nadtan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3148/ : FAILURE in 13m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3902/ : SUCCESS in 13s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1674/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1039/ : UNSTABLE in 16m 32s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3903/ : SUCCESS in 10s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3234/ : SUCCESS in 11m 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4006/ : SUCCESS in 11s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1729/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1080/ : UNSTABLE in 25m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4007/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3236/ : SUCCESS in 5m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4009/ : SUCCESS in 9s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1730/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1081/ : SUCCESS in 20m 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4010/ : SUCCESS in 10s

**Comment by nadtan** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-905.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1082/ : FAILURE in 36m 04s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4011/ : SUCCESS in 8s

**Comment by nadtan** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-905.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1083/ : SUCCESS in 19m 45s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4018/ : SUCCESS in 17s

**Comment by naibinh** [comment]
LGTM

**Comment by nadtan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3243/ : SUCCESS in 13m 46s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4022/ : SUCCESS in 1m 13s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1734/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1087/ : UNSTABLE in 35m 50s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4023/ : SUCCESS in 11s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3246/ : SUCCESS in 4m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4025/ : SUCCESS in 11s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1736/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1088/ : UNSTABLE in 19m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4026/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3249/ : SUCCESS in 2m 49s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4028/ : SUCCESS in 13s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1738/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1089/ : SUCCESS in 21m 04s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4029/ : SUCCESS in 7s

File: src/components/visualizations/chart/legend/LegendItem.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ export default class LegendItem extends React.Component<any, any> {
         const enableBorderRadius = isLineChart(itemChartType) || isAreaChart(itemChartType);
 
         const iconStyle = {
-            borderRadius: enableBorderRadius ? "50%" : "none",
+            borderRadius: enableBorderRadius ? "50%" : "0",
```
Comment (nadtan) [inline src/components/visualizations/chart/legend/LegendItem.tsx]: That's a workaround solution. There is an issue when switching chart type from "line"/"area" -> "column", the border radius of legend icon is not updated (from "50%" -> "none")

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3257/ : FAILURE in 3m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4038/ : SUCCESS in 18s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3264/ : SUCCESS in 4m 08s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4043/ : SUCCESS in 7s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1750/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1097/ : SUCCESS in 19m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4044/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1759/ : SUCCESS in 19m 51s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3283/ : SUCCESS in 5m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/898/ : SUCCESS in 35s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/930/ : SUCCESS in 15m 42s

**Pull request merged**